src/validation.cpp

```diff
@@ -1159,6 +1159,24 @@ CTransactionRef GetTransaction(const CBlockIndex* const block_index, const CTxMe
 {
     LOCK(cs_main);
 
+    if (mempool) {
+        CTransactionRef ptx = mempool->get(hash);
+        if (ptx) {
+            // Transaction was found in mempool but we were searching for it in a block.
```

Comment:
It could be an orphaned block and tx went back to the pool?

Reply:
👍🏼